pkg/clients/rds/rds.go

```diff
@@ -452,7 +458,59 @@ func IsUpToDate(p v1beta1.RDSInstanceParameters, db rds.DBInstance) (bool, error
 		cmpopts.IgnoreFields(v1beta1.RDSInstanceParameters{}, "Tags"),
 		cmpopts.IgnoreFields(v1beta1.RDSInstanceParameters{}, "SkipFinalSnapshotBeforeDeletion"),
 		cmpopts.IgnoreFields(v1beta1.RDSInstanceParameters{}, "FinalDBSnapshotIdentifier"),
-	), nil
+		cmpopts.IgnoreFields(v1beta1.RDSInstanceParameters{}, "ApplyModificationsImmediately"),
+		cmpopts.IgnoreFields(v1beta1.RDSInstanceParameters{}, "AllowMajorVersionUpgrade"),
+		cmpopts.IgnoreFields(v1beta1.RDSInstanceParameters{}, "MasterPasswordSecretRef"),
+	) && !pwdChanged, nil
+}
+
+// GetPassword fetches the referenced input password for an RDSInstance CRD and determines whether it has changed or not
+func GetPassword(ctx context.Context, kube *client.Client, inputCreds *v1alpha1.SecretKeySelector, outCreds *v1alpha1.SecretReference) (string, bool, error) {
+	if kube == nil {
+		return "", false, nil
+	}
+	if inputCreds == nil && outCreds == nil {
+		return "", false, nil
+	}
+
+	var err error
+	var newPwd string
+	var pwdChanged bool
+	if inputCreds != nil {
```

Comment:
I believe once we accept `*v1beta1.RDSInstance` and do the necessary checks like `if rds.Spec.ForProvider.MasterPasswordSecretRef == nil` in the beginning, we don't need to check them anymore. Once they are out of the way, it could be a simple flow like `do the checks -> get the input secret -> get the output secret -> return newPwd, newPwd != oldPwd, nil`

Reply:
I basically just exit early if neither are set. It's kind of a redundant conditional upon further inspection.

I need to check each one specifically, because otherwise accessing the name, namespace and key (for input) will throw an error. If you take a look at the updated function and can figure out a less verbose way to write it and cover the cases it does I'll happily make changes!

Reply:
I left some suggestions about this.